src/profile-logic/import/chrome.js

```diff
@@ -0,0 +1,363 @@
+            'Unable to find the prefix stack index from a node index.'
+          );
+        }
+        frameTable.address[frameIndex] = stringTable.indexForString('');
+        frameTable.category[frameIndex] = javascriptCategoryIndex;
+        frameTable.func[frameIndex] = funcId;
+        frameTable.implementation[frameIndex] = null;
+        frameTable.line[frameIndex] =
+          lineNumber === undefined ? null : lineNumber;
+        frameTable.column[frameIndex] =
+          columnNumber === undefined ? null : columnNumber;
+        frameTable.optimizations[frameIndex] = null;
+        frameTable.length = Math.max(frameTable.length, frameIndex + 1);
+
+        stackTable.frame.push(frameIndex);
+        stackTable.category.push(javascriptCategoryIndex);
+        stackTable.prefix.push(prefixStackIndex);
+        nodeIdToStackId.set(nodeIndex, stackTable.length++);
+      }
+    }
+
+    // Chrome profiles sample much more frequently than Gecko ones do, and they store
+    // the time delta between each sampling event. In order to properly reconstruct
+    // the data using our fixed-time intervals, sample the data at a fixed rate that
+    // is most likely slightly higher. Chrome profiles have been observed sampling
+    // between 100us to 300us. Reconstruct the profile at 500us, which is a somewhat
+    // reasonable interval.
+
+    // Choose 500us as a somewhat reasonable sampling interval. When converting
+    // the chrome profile, this function samples the chrome profile, and generates
+    // new samples on our target interval of 500us.
+    profile.meta.interval = 0.5;
+
+    for (let i = 0; i < samples.length; i++) {
+      const nodeIndex = samples[i];
+      // Convert to milliseconds:
+      threadInfo.lastSeenTime += timeDeltas[i] / 1000;
+      if (
+        threadInfo.lastSeenTime - threadInfo.lastSampledTime >=
+        profile.meta.interval
+      ) {
+        threadInfo.lastSampledTime = threadInfo.lastSeenTime;
+        const stackIndex = ensureExists(
+          nodeIdToStackId.get(nodeIndex),
+          'Could not find the stack information for a sample when decoding a Chrome profile.'
+        );
+        samplesTable.responsiveness.push(null);
+        samplesTable.stack.push(stackIndex);
+        samplesTable.time.push(threadInfo.lastSampledTime);
+        samplesTable.rss.push(null);
+        samplesTable.uss.push(null);
+        samplesTable.length++;
+      }
+    }
+  }
+
+  for (const thread of profile.threads) {
+    assertStackOrdering(thread.stackTable);
+  }
+
+  await extractScreenshots(profile, (eventsByName.get('Screenshot'): any));
+
+  return profile;
+}
+
+async function extractScreenshots(
+  profile: Profile,
+  screenshots: ?(ScreenshotEvent[])
+): Promise<void> {
+  if (!screenshots) {
+    return;
+  }
+  const thread = profile.threads[0];
+  if (!thread) {
+    throw new Error('Expected to find a thread to attach the screenshots to.');
+  }
+
+  const imageElement = new Image();
```

Comment:
I think we are doing this calculation inside `getImageSize` now. It looks like we can delete that variable.

Reply:
Hah! That would explain the NaN errors from the component.